tools/src/bin/build-blame.rs

```diff
@@ -0,0 +1,357 @@
+extern crate env_logger;
+extern crate git2;
+#[macro_use]
+extern crate log;
+extern crate tools;
+extern crate unicode_normalization;
+
+use std::borrow::Borrow;
+use std::collections::HashMap;
+use std::env;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+use git2::{DiffFindOptions, ObjectType, Oid, Patch, Repository, Sort};
+use tools::config::index_blame;
+use tools::git_ops;
+use unicode_normalization::UnicodeNormalization;
+
+fn get_hg_rev(git_repo: &Repository, git_oid: &Oid) -> Option<String> {
+    // TODO: start a helper process and stream I/O to it instead of spawning a new process every time
+    let output = Command::new("git")
+        .arg("cinnabar")
+        .arg("git2hg")
+        .arg(format!("{}", git_oid))
+        .current_dir(git_repo.path())
+        .output()
+        .unwrap();
+    if !output.status.success() {
+        panic!("`git cinnabar git2hg {}` returned error code {:?}", git_oid, output.status.code());
+    }
+    match git_ops::decode_bytes(output.stdout).lines().next() {
```

Comment:
You can:

```
let line = git_ops::decode_bytes(output.stdout).lines().next()?;
if line != ... {
    return None;
}
Some(line.to_string())
```

Reply:
This code goes away in the next patch so I ignored this.